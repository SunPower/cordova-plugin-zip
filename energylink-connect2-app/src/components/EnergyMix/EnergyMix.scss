@import '~@sunpower/theme-dark/styles/variables';

.energy-mix-container {
  .title {
    color: $white;
  }

  .subtitle {
    color: $white;
  }

  .percentage {
    text-align: center;
  }

  .energy-mix-chart {
    margin: auto;
    .home-usage-label {
      &-title {
        fill: $white;
        font-size: $font-size;
      }
      &-info {
        fill: $white;
        font-weight: $bold-text;
        font-size: $font-size-big;
      }
    }
  }

  .energy-source {
    flex-direction: column;
    justify-content: center;
    align-items: center;
    &-label {
      font-size: $font-size-medium;
      color: $white;
      text-align: center;
      max-width: 50px;
      &.battery {
        max-width: 70px;
      }
      &.grid {
        max-width: 45px;
      }
    }
    &-info {
      font-size: $font-size-big;
      text-align: center;
      font-weight: $bold-text;
      &.yellow {
        color: $daisy;
      }
      &.orange {
        color: $sun;
      }
      &.blue {
        color: $blue-default;
      }
    }
  }

  .icons {
    align-items: center;
    justify-content: flex-end;
    text-align: center;
  }
}
