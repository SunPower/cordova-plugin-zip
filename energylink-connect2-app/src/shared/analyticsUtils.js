import {
  __,
  compose,
  curry,
  gt,
  isEmpty,
  join,
  lensIndex,
  map,
  not,
  over,
  props,
  propSatisfies,
  toLower,
  toUpper,
  when
} from 'ramda'

export const isNotEmpty = compose(not, isEmpty)

export const capitalizeWord = when(
  propSatisfies(gt(__, 0), 'length'),
  compose(join(''), over(lensIndex(0), toUpper), toLower)
)

const stringValueOrEmptyString = stringValue => stringValue || ''

const getStringProps = curry(compose(map(stringValueOrEmptyString), props))

export const getUserProfile = getStringProps([
  'uniqueId',
  'firstName',
  'lastName',
  'email',
  'dealerName',
  'recordType'
])
