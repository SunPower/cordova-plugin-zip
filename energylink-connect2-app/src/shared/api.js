import Swagger from 'swagger-client'
let apiParty, apiSite, apiFirmware, apiScanSN

const requestOptions = access_token => ({
  requestInterceptor: req => {
    if (access_token) req.headers['Authorization'] = `Bearer ${access_token}`
    return req
  }
})

export async function getApiFirmware() {
  if (!apiFirmware)
    apiFirmware = await Swagger(process.env.REACT_APP_SWAGGER_FIRMWARE)
  return apiFirmware
}
export async function getApiPVS() {
  return await Swagger(process.env.REACT_APP_PVS_ADDRESS)
}
export async function getApiDevice(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_DEVICE,
    requestOptions(access_token)
  )
}
export async function getApiParty(access_token) {
  apiParty = await Swagger(
    process.env.REACT_APP_SWAGGER_PARTY,
    requestOptions(access_token)
  )
  return apiParty
}
export async function getApiSearch(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_SEARCH,
    requestOptions(access_token)
  )
}
export async function getApiSite(access_token) {
  if (!apiSite)
    apiSite = await Swagger(
      process.env.REACT_APP_SWAGGER_SITE,
      requestOptions(access_token)
    )
  return apiSite
}
export async function getApiScanSN() {
  if (!apiScanSN)
    apiScanSN = await Swagger(process.env.REACT_APP_SCAN_SERIAL_NUMBERS)
  return apiScanSN
}
export async function getApiAuth(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_AUTH,
    requestOptions(access_token)
  )
}
export const storageSwaggerTag = 'Commissioning'
