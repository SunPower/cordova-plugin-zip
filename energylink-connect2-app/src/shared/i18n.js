import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
