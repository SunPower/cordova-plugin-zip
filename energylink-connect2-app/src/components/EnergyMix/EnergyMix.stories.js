import React from 'react'

import { storiesOf } from '@storybook/react'

import EnergyMix from './index'

storiesOf('EnergyMix', module).add('Simple', () => (
  <EnergyMix
    data={{
      date: new Date(2019, 9, 9, 12, 0, 0),
      solar: 2.3,
      storage: 1,
      grid: 1,
      homeUsage: 4.3
    }}
  />
))
