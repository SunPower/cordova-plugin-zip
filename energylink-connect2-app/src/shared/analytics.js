import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
