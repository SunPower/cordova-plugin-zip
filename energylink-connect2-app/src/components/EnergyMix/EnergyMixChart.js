import React from 'react'
import { PieChart, Pie, Cell, Label } from 'recharts'

import { useI18n } from '../../shared/i18n'
import { Home } from '../Icons'

const COLORS = ['#2B93CC', '#F7921E', '#FFE600']

const renderHome = props => {
  const {
    t,
    homeUsage,
    viewBox: { cx, cy }
  } = props

  return (
    <svg
      className="home-usage-label"
      width={140}
      height={140}
      x={cx - 70}
      y={cy - 50}
    >
      <Home x={39} />
      <text
        x={70}
        y={65}
        textAnchor="middle"
        fill="#FFFFFF"
        className="home-usage-label-title is-uppercase"
      >
        {t('HOME_USAGE')}
      </text>
      <text x={70} y={85} textAnchor="middle" className="home-usage-label-info">
        {homeUsage} {t('KWH')}
      </text>
    </svg>
  )
}
export default function EnergyMix({
  storage = 0,
  grid = 0,
  solar = 0,
  homeUsage = 0,
  isAnimated = true
}) {
  const t = useI18n()
  const chartData = [
    { value: grid < 0 ? 0 : grid },
    { value: storage < 0 ? 0 : storage },
    { value: solar }
  ]

  return (
    <PieChart className="energy-mix-chart" width={195} height={195}>
      <Pie
        data={chartData}
        innerRadius={83}
        outerRadius={95}
        dataKey="value"
        blendStroke={true}
        startAngle={-180}
        endAngle={360}
        isAnimationActive={isAnimated}
      >
        {chartData.map((entry, index) => (
          <Cell
            key={`cell-${index}`}
            fill={COLORS[index % COLORS.length]}
            filter="url(#filter0_i)"
          />
        ))}
        <Label
          offset={0}
          position="center"
          t={t}
          homeUsage={homeUsage.toFixed(2)}
          content={renderHome}
        />
      </Pie>
      <defs>
        <filter
          id="filter0_i"
          x="0.5"
          y="0.807617"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix" />
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          />
          <feColorMatrix
            in="SourceAlpha"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            result="hardAlpha"
          />
          <feOffset dy="4" />
          <feGaussianBlur stdDeviation="2" />
          <feComposite in2="hardAlpha" operator="arithmetic" k2="-1" k3="1" />
          <feColorMatrix
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
          />
          <feBlend mode="normal" in2="shape" result="effect1_innerShadow" />
        </filter>
      </defs>
    </PieChart>
  )
}
