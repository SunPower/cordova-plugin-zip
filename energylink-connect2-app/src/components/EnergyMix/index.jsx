import React from 'react'
import EnergyMixChart from './EnergyMixChart'
import { Battery, Grid, SolarPanel } from '../Icons'
import { useI18n } from 'shared/i18n'
import './EnergyMix.scss'

export default function EnergyMix(props) {
  const { solar = 0, storage = 0, grid = 0, homeUsage = 0 } = props
  const t = useI18n()
  const solarPercentage =
    solar !== 0
      ? (Math.min(solar, homeUsage) * 100) / Math.max(solar, homeUsage)
      : 0
  return (
    <div className="energy-mix-container columns  is-mobile is-multiline is-variable is-1">
      <div className="column is-full">
        <h6 className="percentage is-uppercase is-6 mb-10">
          {t('SOLAR')} <b>{solarPercentage.toFixed(2)}%</b>
        </h6>
        <EnergyMixChart {...props} />
      </div>
      <div className="column is-4 is-flex energy-source">
        <SolarPanel />
        <p className="is-uppercase energy-source-label solar mt-8">
          {t('FROM')} {t('SOLAR')}
        </p>
        <p className="energy-source-info yellow mt-5">
          {solar.toFixed(2)} <br /> {t('KWH')}
        </p>
      </div>
      <div className="column is-4 is-flex energy-source">
        <Battery />
        <p className="is-uppercase energy-source-label battery mt-8">
          {storage < 0 ? t('TO') : t('FROM')} {t('BATTERY')}
        </p>
        <p className="energy-source-info orange mt-5">
          {Math.abs(storage).toFixed(2)} <br /> {t('KWH')}
        </p>
      </div>
      <div className="column is-4 is-flex energy-source">
        <Grid />
        <p className="is-uppercase energy-source-label grid mt-8">
          {grid < 0 ? t('TO') : t('FROM')} {t('GRID')}
        </p>
        <p className="energy-source-info blue mt-5">
          {Math.abs(grid).toFixed(2)} <br /> {t('KWH')}
        </p>
      </div>
    </div>
  )
}
