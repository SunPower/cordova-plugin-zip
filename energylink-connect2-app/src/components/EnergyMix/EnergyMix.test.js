import { shallow } from 'enzyme'
import moment from 'moment'
import React from 'react'
import * as reactRedux from 'react-redux'

import * as i18n from '../../shared/i18n'

import EnergyMixChart from './EnergyMixChart'

import EnergyMix from './index'

describe('EnergyMix component', () => {
  beforeEach(() => {
    let dispatchMock = jest.fn()
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
    jest
      .spyOn(i18n, 'useI18nComponent')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => dispatchMock)
  })

  it('renders properly', () => {
    const component = shallow(
      <EnergyMix
        date={moment('2019-10-09T11:00:00.000Z')}
        solar={2.3}
        storage={1}
        grid={1}
        homeUsage={4.3}
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('renders the energy mix chart', () => {
    const component = shallow(
      <EnergyMix
        date={moment('2019-10-09T11:00:00.000Z')}
        solar={2.3}
        storage={1}
        grid={1}
        homeUsage={4.3}
      />
    )
    expect(component.find(EnergyMixChart).dive()).toMatchSnapshot()
  })
})
