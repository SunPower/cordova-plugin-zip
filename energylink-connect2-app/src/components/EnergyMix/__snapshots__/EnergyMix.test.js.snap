// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnergyMix component renders properly 1`] = `
<div
  className="energy-mix-container columns  is-mobile is-multiline is-variable is-1"
  id="energy-mix-container"
>
  <div
    className="column is-8"
  >
    <h6
      className="title is-uppercase is-6"
    >
      TODAYS_ENERGY_MIX
    </h6>
    <h6
      className="subtitle is-uppercase is-6"
    >
      AS_OF_TIME_TODAY 04:00AM
    </h6>
  </div>
  <div
    className="column is-full"
  >
    <h6
      className="percentage is-uppercase is-6 mb-10"
    >
      SOLAR
       
      <b>
        53.49
        %
      </b>
    </h6>
    <EnergyMix
      date={"2019-10-09T11:00:00.000Z"}
      grid={1}
      homeUsage={4.3}
      solar={2.3}
      storage={1}
    />
  </div>
  <div
    className="column is-4 is-flex energy-source"
  >
    <ForwardRef(SvgSolarPanel) />
    <p
      className="is-uppercase energy-source-label solar mt-8"
    >
      FROM
       
      SOLAR
    </p>
    <p
      className="energy-source-info yellow mt-5"
    >
      2.30
       
      <br />
       
      KWH
    </p>
  </div>
  <div
    className="column is-4 is-flex energy-source"
  >
    <ForwardRef(SvgBattery) />
    <p
      className="is-uppercase energy-source-label battery mt-8"
    >
      FROM
       
      BATTERY
    </p>
    <p
      className="energy-source-info orange mt-5"
    >
      1.00
       
      <br />
       
      KWH
    </p>
  </div>
  <div
    className="column is-4 is-flex energy-source"
  >
    <ForwardRef(SvgGrid) />
    <p
      className="is-uppercase energy-source-label grid mt-8"
    >
      FROM
       
      GRID
    </p>
    <p
      className="energy-source-info blue mt-5"
    >
      1.00
       
      <br />
       
      KWH
    </p>
  </div>
</div>
`;

exports[`EnergyMix component renders the energy mix chart 1`] = `
<PieChart
  barCategoryGap="10%"
  barGap={4}
  className="energy-mix-chart"
  cx="50%"
  cy="50%"
  endAngle={360}
  height={195}
  innerRadius={0}
  layout="centric"
  margin={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  outerRadius="80%"
  reverseStackOrder={false}
  stackOffset="none"
  startAngle={0}
  width={195}
>
  <Pie
    animationBegin={400}
    animationDuration={1500}
    animationEasing="ease"
    blendStroke={true}
    cx="50%"
    cy="50%"
    data={
      Array [
        Object {
          "value": 1,
        },
        Object {
          "value": 1,
        },
        Object {
          "value": 2.3,
        },
      ]
    }
    dataKey="value"
    endAngle={360}
    fill="#808080"
    hide={false}
    innerRadius={83}
    isAnimationActive={true}
    labelLine={true}
    legendType="rect"
    minAngle={0}
    nameKey="name"
    outerRadius={95}
    paddingAngle={0}
    startAngle={-180}
    stroke="#fff"
  >
    <Cell
      fill="#2B93CC"
      filter="url(#filter0_i)"
      key="cell-0"
    />
    <Cell
      fill="#F7921E"
      filter="url(#filter0_i)"
      key="cell-1"
    />
    <Cell
      fill="#FFE600"
      filter="url(#filter0_i)"
      key="cell-2"
    />
    <Label
      content={[Function]}
      homeUsage="4.30"
      offset={0}
      position="center"
      t={[Function]}
    />
  </Pie>
  <defs>
    <filter
      colorInterpolationFilters="sRGB"
      filterUnits="userSpaceOnUse"
      id="filter0_i"
      x="0.5"
      y="0.807617"
    >
      <feFlood
        floodOpacity="0"
        result="BackgroundImageFix"
      />
      <feBlend
        in="SourceGraphic"
        in2="BackgroundImageFix"
        mode="normal"
        result="shape"
      />
      <feColorMatrix
        in="SourceAlpha"
        result="hardAlpha"
        type="matrix"
        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
      />
      <feOffset
        dy="4"
      />
      <feGaussianBlur
        stdDeviation="2"
      />
      <feComposite
        in2="hardAlpha"
        k2="-1"
        k3="1"
        operator="arithmetic"
      />
      <feColorMatrix
        type="matrix"
        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
      />
      <feBlend
        in2="shape"
        mode="normal"
        result="effect1_innerShadow"
      />
    </filter>
  </defs>
</PieChart>
`;
